src/content/docs/dev/general-message-passing/cosmos-gmp/cosmos-2way-manual-relay.mdx

```diff
@@ -53,11 +54,11 @@ ethereum \\
 
 ### 2. Monitor for the `MessageReceived` Event
 
-After confirming the gateway transaction, wait for the `MessageReceived` event on the Axelar network. This event indicates that the message from the source chain has been received and is ready to be routed.
+After confirming the gateway transaction, wait for the [`MessageReceived`](https://github.com/axelarnetwork/axelar-core/blob/4e67aa564db2b10eec25831ed1146c94a9172c60/proto/axelar/nexus/v1beta1/events.proto#L49) event on the Axelar network. This event indicates that the message from the source chain has been received and is ready to be routed.
```

Comment:
why link to this commit hash?

Reply:
I mean wouldnt it be better to link to specific release tag?
https://github.com/axelarnetwork/axelar-core/blob/v1.2.1/proto/axelar/nexus/v1beta1/events.proto#L49

or main branch

Reply:
We can't use a release that would be confusing especially if we have more release that has overitten that implementation.

However, I think the main branch is the way to go. Nevertheless, you can't copy the permalink without the blob, which I don't find a bad idea. We already have it documented like that.